File: Data Structures & Algorithms/valid-parenthesis-string/submission-3.py
class Solution:
    def checkValidString(self, s: str) -> bool:
        min_open = max_open = 0
        for char in s:
            if char == "(":
                min_open += 1
                max_open += 1
            else:
                min_open = max(0,min_open-1)
                if char == ")":
                    max_open -= 1
                else:
                    max_open += 1

            if max_open < 0:
                return False

        return min_open == 0

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        var hSet = new HashSet<int>(nums);
        int res = 0;

        foreach (int num in hSet){
            if (!hSet.Contains(num-1)){
                int currNum = num;
                int length = 1;

                while (hSet.Contains(currNum+1)){
                    currNum++;
                    length++;
                }

                res = Math.Max(res, length);
            }
        }

        return res;
    }
}
